src/proof/theory_proof.cpp

```diff
@@ -1264,4 +1263,387 @@ void TheoryProof::printRewriteProof(std::ostream& os, const Node &n1, const Node
   os << "))";
 }
 
+
+inline bool TheoryProof::match(TNode n1, TNode n2, theory::TheoryId theoryId)
+{
+  bool ufProof = (theoryId == theory::THEORY_UF);
+  Debug(ufProof ? "pf::uf" : "mgd") << "match " << n1 << " " << n2 << std::endl;
+  if (ProofManager::currentPM()->hasOp(n1))
```

Comment:
We should get the pointer to `ProofManager::currentPM()` once at the beginning of the function to make this code more readable. Maybe in another PR.